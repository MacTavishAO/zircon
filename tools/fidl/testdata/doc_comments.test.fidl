/// library comment #1
///
/// library comment #2
library test.name;


/// bits comment #1
///
/// bits comment #2
bits MyBits:uint32 {
    /// bits member comment #1
    ///
    /// bits member comment #3
    MY_FIRST_BIT = 1;
    MY_OTHER_BIT = 2;
};


/// const comment #1
///
/// const comment #3
const int32 C = 4;


/// enum comment #1.
///
/// enum comment #2.
enum MyEnum : uint32 {
    /// enum Member comment 1.
    FOO = 1;
    // Internal comment.
    BAR = 2;
};

/// struct comment #1
///
/// struct comment #3
struct Struct {
    /// struct member comment #1
    ///
    /// struct member comment #3
    int32 Field;
};

/// union comment #1
///
/// union comment #3
union Union {
    /// union member comment #1
    ///
    /// union member comment #3
    1: int32 Field;
};

/// table comment #1
///
/// table comment #3
table Table {
    /// table field comment #1
    ///
    /// table field comment #3
    1: int32 Field;
};

/// interface comment #1
///
/// interface comment #3
protocol Interface {
    /// method comment #1
    ///
    /// method comment #3
    Method();

    /// event comment #1
    ///
    /// event comment #3
    -> OnEvent();
};

/// service comment #1
///
/// service comment #3
service Service {
    /// member comment #1
    ///
    /// member comment #3
    Interface interface;
};
