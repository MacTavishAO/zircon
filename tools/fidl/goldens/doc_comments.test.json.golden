{
  "version": "0.0.1",
  "name": "test.name",
  "maybe_attributes": [
    {
      "name": "Doc",
      "value": " library comment #1\n\n library comment #2\n"
    }
  ],
  "library_dependencies": [],
  "bits_declarations": [
    {
      "name": "test.name/MyBits",
      "location": {
        "filename": "doc_comments.test.fidl",
        "line": 10,
        "column": 6,
        "length": 6
      },
      "maybe_attributes": [
        {
          "name": "Doc",
          "value": " bits comment #1\n\n bits comment #2\n"
        }
      ],
      "type": {
        "kind": "primitive",
        "subtype": "uint32"
      },
      "mask": "3",
      "members": [
        {
          "name": "MY_FIRST_BIT",
          "location": {
            "filename": "doc_comments.test.fidl",
            "line": 14,
            "column": 5,
            "length": 12
          },
          "value": {
            "kind": "literal",
            "value": "1",
            "expression": "1",
            "literal": {
              "kind": "numeric",
              "value": "1",
              "expression": "1"
            }
          },
          "maybe_attributes": [
            {
              "name": "Doc",
              "value": " bits member comment #1\n\n bits member comment #3\n"
            }
          ]
        },
        {
          "name": "MY_OTHER_BIT",
          "location": {
            "filename": "doc_comments.test.fidl",
            "line": 15,
            "column": 5,
            "length": 12
          },
          "value": {
            "kind": "literal",
            "value": "2",
            "expression": "2",
            "literal": {
              "kind": "numeric",
              "value": "2",
              "expression": "2"
            }
          }
        }
      ],
      "strict": true
    }
  ],
  "const_declarations": [
    {
      "name": "test.name/C",
      "location": {
        "filename": "doc_comments.test.fidl",
        "line": 22,
        "column": 13,
        "length": 1
      },
      "maybe_attributes": [
        {
          "name": "Doc",
          "value": " const comment #1\n\n const comment #3\n"
        }
      ],
      "type": {
        "kind": "primitive",
        "subtype": "int32"
      },
      "value": {
        "kind": "literal",
        "value": "4",
        "expression": "4",
        "literal": {
          "kind": "numeric",
          "value": "4",
          "expression": "4"
        }
      }
    }
  ],
  "enum_declarations": [
    {
      "name": "test.name/MyEnum",
      "location": {
        "filename": "doc_comments.test.fidl",
        "line": 28,
        "column": 6,
        "length": 6
      },
      "maybe_attributes": [
        {
          "name": "Doc",
          "value": " enum comment #1.\n\n enum comment #2.\n"
        }
      ],
      "type": "uint32",
      "members": [
        {
          "name": "FOO",
          "location": {
            "filename": "doc_comments.test.fidl",
            "line": 30,
            "column": 5,
            "length": 3
          },
          "value": {
            "kind": "literal",
            "value": "1",
            "expression": "1",
            "literal": {
              "kind": "numeric",
              "value": "1",
              "expression": "1"
            }
          },
          "maybe_attributes": [
            {
              "name": "Doc",
              "value": " enum Member comment 1.\n"
            }
          ]
        },
        {
          "name": "BAR",
          "location": {
            "filename": "doc_comments.test.fidl",
            "line": 32,
            "column": 5,
            "length": 3
          },
          "value": {
            "kind": "literal",
            "value": "2",
            "expression": "2",
            "literal": {
              "kind": "numeric",
              "value": "2",
              "expression": "2"
            }
          }
        }
      ],
      "strict": true
    }
  ],
  "experimental_resource_declarations": [],
  "interface_declarations": [
    {
      "name": "test.name/Interface",
      "location": {
        "filename": "doc_comments.test.fidl",
        "line": 68,
        "column": 10,
        "length": 9
      },
      "maybe_attributes": [
        {
          "name": "Doc",
          "value": " interface comment #1\n\n interface comment #3\n"
        }
      ],
      "methods": [
        {
          "ordinal": 6982550709377523448,
          "name": "Method",
          "location": {
            "filename": "doc_comments.test.fidl",
            "line": 72,
            "column": 5,
            "length": 6
          },
          "has_request": true,
          "maybe_attributes": [
            {
              "name": "Doc",
              "value": " method comment #1\n\n method comment #3\n"
            }
          ],
          "maybe_request": [],
          "maybe_request_type_shape_v1": {
            "inline_size": 16,
            "alignment": 8,
            "depth": 0,
            "max_handles": 0,
            "max_out_of_line": 0,
            "has_padding": false,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "has_response": false,
          "is_composed": false
        },
        {
          "ordinal": 654361314665317613,
          "name": "OnEvent",
          "location": {
            "filename": "doc_comments.test.fidl",
            "line": 77,
            "column": 8,
            "length": 7
          },
          "has_request": false,
          "maybe_attributes": [
            {
              "name": "Doc",
              "value": " event comment #1\n\n event comment #3\n"
            }
          ],
          "has_response": true,
          "maybe_response": [],
          "maybe_response_type_shape_v1": {
            "inline_size": 16,
            "alignment": 8,
            "depth": 0,
            "max_handles": 0,
            "max_out_of_line": 0,
            "has_padding": false,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "is_composed": false
        }
      ]
    }
  ],
  "service_declarations": [
    {
      "name": "test.name/Service",
      "location": {
        "filename": "doc_comments.test.fidl",
        "line": 83,
        "column": 9,
        "length": 7
      },
      "maybe_attributes": [
        {
          "name": "Doc",
          "value": " service comment #1\n\n service comment #3\n"
        }
      ],
      "members": [
        {
          "type": {
            "kind": "identifier",
            "identifier": "test.name/Interface",
            "nullable": false
          },
          "name": "interface",
          "location": {
            "filename": "doc_comments.test.fidl",
            "line": 87,
            "column": 15,
            "length": 9
          },
          "maybe_attributes": [
            {
              "name": "Doc",
              "value": " member comment #1\n\n member comment #3\n"
            }
          ]
        }
      ]
    }
  ],
  "struct_declarations": [
    {
      "name": "test.name/Struct",
      "location": {
        "filename": "doc_comments.test.fidl",
        "line": 38,
        "column": 8,
        "length": 6
      },
      "anonymous": false,
      "maybe_attributes": [
        {
          "name": "Doc",
          "value": " struct comment #1\n\n struct comment #3\n"
        }
      ],
      "members": [
        {
          "type": {
            "kind": "primitive",
            "subtype": "int32"
          },
          "name": "Field",
          "location": {
            "filename": "doc_comments.test.fidl",
            "line": 42,
            "column": 11,
            "length": 5
          },
          "maybe_attributes": [
            {
              "name": "Doc",
              "value": " struct member comment #1\n\n struct member comment #3\n"
            }
          ],
          "field_shape_v1": {
            "offset": 0,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 4,
        "alignment": 4,
        "depth": 0,
        "max_handles": 0,
        "max_out_of_line": 0,
        "has_padding": false,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    }
  ],
  "table_declarations": [
    {
      "name": "test.name/Table",
      "location": {
        "filename": "doc_comments.test.fidl",
        "line": 58,
        "column": 7,
        "length": 5
      },
      "maybe_attributes": [
        {
          "name": "Doc",
          "value": " table comment #1\n\n table comment #3\n"
        }
      ],
      "members": [
        {
          "ordinal": 1,
          "reserved": false,
          "type": {
            "kind": "primitive",
            "subtype": "int32"
          },
          "name": "Field",
          "location": {
            "filename": "doc_comments.test.fidl",
            "line": 62,
            "column": 14,
            "length": 5
          },
          "maybe_attributes": [
            {
              "name": "Doc",
              "value": " table field comment #1\n\n table field comment #3\n"
            }
          ]
        }
      ],
      "strict": false,
      "type_shape_v1": {
        "inline_size": 16,
        "alignment": 8,
        "depth": 2,
        "max_handles": 0,
        "max_out_of_line": 24,
        "has_padding": true,
        "has_flexible_envelope": true,
        "is_resource": false
      }
    }
  ],
  "union_declarations": [
    {
      "name": "test.name/Union",
      "location": {
        "filename": "doc_comments.test.fidl",
        "line": 48,
        "column": 7,
        "length": 5
      },
      "maybe_attributes": [
        {
          "name": "Doc",
          "value": " union comment #1\n\n union comment #3\n"
        }
      ],
      "members": [
        {
          "ordinal": 1,
          "reserved": false,
          "name": "Field",
          "type": {
            "kind": "primitive",
            "subtype": "int32"
          },
          "location": {
            "filename": "doc_comments.test.fidl",
            "line": 52,
            "column": 14,
            "length": 5
          },
          "maybe_attributes": [
            {
              "name": "Doc",
              "value": " union member comment #1\n\n union member comment #3\n"
            }
          ]
        }
      ],
      "strict": true,
      "type_shape_v1": {
        "inline_size": 24,
        "alignment": 8,
        "depth": 1,
        "max_handles": 0,
        "max_out_of_line": 8,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    }
  ],
  "type_alias_declarations": [],
  "declaration_order": [
    "test.name/Union",
    "test.name/Table",
    "test.name/Struct",
    "test.name/Interface",
    "test.name/Service",
    "test.name/MyEnum",
    "test.name/MyBits",
    "test.name/C"
  ],
  "declarations": {
    "test.name/MyBits": "bits",
    "test.name/C": "const",
    "test.name/MyEnum": "enum",
    "test.name/Interface": "interface",
    "test.name/Service": "service",
    "test.name/SomeLongAnonymousPrefix0": "struct",
    "test.name/SomeLongAnonymousPrefix1": "struct",
    "test.name/Struct": "struct",
    "test.name/Table": "table",
    "test.name/Union": "union"
  }
}
